app/models/client_notification.rb

```diff
@@ -0,0 +1,14 @@
+class ClientNotification < ApplicationRecord
+
+  before_save :ensure_exactly_one_active
```

Comment:
This should be called in `after_save`, i.e. the other records should only be modified once the active `ClientNotification` has been successfully saved.

Though in practice it doesn't matter much since the saving and before/after callbacks are all run in one DB transaction (except for `after_commit`), so if anything goes wrong all changes will be rolled back.

(so just FYI, no need to really change it I think)

Reply:
👍